clippy_lints/src/erasing_op.rs

```diff
@@ -0,0 +1,62 @@
+use consts::{constant_simple, Constant};
+use rustc::hir::*;
+use rustc::lint::*;
+use syntax::codemap::Span;
+use utils::{in_macro, span_lint};
+
+/// **What it does:** Checks for erasing operations, e.g. `x * 0`.
+///
+/// **Why is this bad?** The whole expression can be replaced by zero.
+/// Most likely mistake was made and code should be reviewed or simplified.
+///
+/// **Known problems:** None.
+///
+/// **Example:**
+/// ```rust
+/// 0 / x; 0 * x; x & 0
+/// ```
+declare_lint! {
+    pub ERASING_OP,
+    Warn,
+    "using erasing operations, e.g. `x * 0` or `y & 0`"
```

Comment:
There is also  `x | -1` (signed) / `x | !0` (unsigned), though those return `-1` and `!0` respectively. It's ok if we extend this later, though.

Reply:
Yes, I can add this lint. We might also change the lint name. Something about expression with a constant result, I'm not sure.